source/core/CEntryPoints.cpp

```diff
@@ -168,39 +168,33 @@ VIREO_EXPORT NIError EggShell_ReadDouble(TypeManagerRef tm, const TypeRef actual
 }
 //------------------------------------------------------------
 // Write a string value to a symbol. Value will be parsed according to format designated.
-VIREO_EXPORT void EggShell_WriteValueString(TypeManagerRef tm,
-        const char* viName, const char* eltName, const char* format, const char* value)
+VIREO_EXPORT EggShellResult EggShell_WriteValueString(TypeManagerRef tm, const TypeRef typeRef, void* data, const char* format, const char* value)
 {
     TypeManagerScope scope(tm);
 
-    void *pData = nullptr;
-
-    SubString objectName(viName);
-    SubString path(eltName);
     SubString valueString(value);
 
-    TypeRef actualType = tm->GetObjectElementAddressFromPath(&objectName, &path, &pData, true);
-    if (actualType == nullptr)
-        return;
+    if (typeRef == nullptr || !typeRef->IsValid())
+        return kEggShellResult_InvalidTypeRef;
 
     EventLog log(EventLog::DevNull);
     SubString formatss(format);
     TDViaParser parser(tm, &valueString, &log, 1, &formatss, true, true, true);
-    parser.ParseData(actualType, pData);
+    Int32 error = parser.ParseData(typeRef, data);
```

Comment:
ParseData gives some pretty good error codes back in the form of LVError + JSONLVError: https://github.com/ni/VireoSDK/blob/master/source/core/TDCodecVia.cpp#L1079

Maybe we should pass that through like how readDouble returns NIError instead of EggShellResult: https://github.com/ni/VireoSDK/blob/api-type-refactor/source/io/module_eggShell.js#L296

Alternatively maybe readDouble should switch to returning EggShellResult. Thoughts?

Reply:
I agree with switching readDouble and writeDouble to EggShellResult. I think that we can use NIError internally for vireo. If we think is needed to preserve more context about the error, we could add more EggShellResult options (which I don't think is needed right now)

Reply:
Sounds good, I'll do that as part of the writeDouble change

Reply:
Agree. Make the API the same. Use EggShellResult consistently. If we need more specific information add error codes to EggShellResult that are more specific.